crates/ruff_linter/src/test.rs

```diff
@@ -90,7 +90,7 @@ pub fn test_snippet(contents: &str, settings: &LinterSettings) -> Vec<Message> {
 }
 
 thread_local! {
-    static MAX_ITERATIONS: std::cell::Cell<usize> = const { std::cell::Cell::new(10) };
+        static MAX_ITERATIONS: std::cell::Cell<usize> = const { std::cell::Cell::new(12) };
```

Comment:
What's the reason for changing this? Can you specify which test case required this?

Reply:
I just ran it on your branch with the original value and it works. I think it can be reverted back.

Reply:
It's very strange but I changed it to the original value and the CI fails:

https://github.com/astral-sh/ruff/actions/runs/10886671657/job/30207264819?pr=11811#step:8:3383

Reply:
I changed it back to original config in this commit but CI fails:
https://github.com/astral-sh/ruff/pull/11811/commits/5079a8c85cce20c47ca6507c50babe74efd0d8b9

https://github.com/astral-sh/ruff/actions/runs/10886671657/job/30207264819?pr=11811#step:8:3383